modules/core/client/components/LanguageSwitch.component.js

```diff
@@ -1,33 +1,153 @@
+import i18n from '@/config/client/i18n';
+import * as usersApi from '@/modules/users/client/api/users.api';
+
+const SelectedLanguage = styled.strong`
+  padding: 7px 13px;
+  font-size: 14px;
+  display: inline-block;
+`;
+
+const LanguageList = styled.ul`
+  margin: 0;
+  padding: 0;
+  list-style: none;
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr;
+  grid-gap: 10px;
+
+  @media (max-width: 480px) {
+    grid-template-columns: 100%;
+  }
+`;
+
+const SearchInput = styled.input`
+  width: 100%;
+  max-width: 250px;
+  margin-bottom: 10px;
+`;
 
 /**
- * The main LanguageSwitch component.
- * @param {'dropdown'|'select'} presentation - should we show dropdown (header), or select (account) selector?
+ * LanguageSwitch component.
+ * @param {String} buttonStyle - Button style: inverse, default, primary
  * @param {Boolean} [saveToAPI=false] - should we save selected language to API?
  */
-export default function LanguageSwitch({
-  presentation = 'dropdown',
-  saveToAPI = false,
-}) {
+export default function LanguageSwitch({ buttonStyle = 'default', saveToAPI }) {
+  const { t } = useTranslation('core');
+  const locales = getLocales();
+  const [search, setSearch] = useState('');
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [currentLanguageCode, setCurrentLanguageCode] = useState(i18n.language);
+
+  const onLanguageChange = async code => {
+    setCurrentLanguageCode(code);
+    i18n.changeLanguage(code);
+
+    // save the user's choice to api
+    if (saveToAPI) {
+      // @TODO this needs some feedback. Currently no feedback to user that this was saved.
+      await usersApi.update({ locale: code });
+    }
+  };
+
+  // selected language
+  const currentLanguage = locales.find(
+    ({ code }) => code === currentLanguageCode,
+  );
+
+  const filteredLocales = search
+    ? getSearchedLocales(locales, search)
+    : locales;
+
+  const onModalHide = () => {
+    setIsModalVisible(false);
+  };
+
   return (
-    <LanguageSwitchContainer
-      presentation={
-        presentation === 'dropdown'
-          ? LanguageSwitchDropdown
-          : LanguageSwitchSelect
-      }
-      saveToAPI={saveToAPI}
-    />
+    <>
+      <button
+        title={currentLanguage?.label}
+        className={classnames('btn', {
+          'btn-primary': buttonStyle === 'primary',
+          'btn-default': buttonStyle === 'default',
+          'btn-inverse': buttonStyle === 'inverse',
+        })}
+        onClick={event => {
+          event.preventDefault();
+          setIsModalVisible(true);
+        }}
+      >
+        {t('Language: {{code}}', {
+          code: currentLanguage?.code?.toUpperCase() ?? 'EN',
+        })}
+      </button>
+      <Modal show={isModalVisible} onHide={onModalHide}>
+        <Modal.Header closeButton>
+          <Modal.Title>{t('Select a language')}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          {locales.length >= 8 && (
```

Comment:
So search input shows up for 8+ languages. Felt like with fewer languages it isn't needed.